pkg/apis/serving/v1alpha1/kfservice_validation.go

```diff
@@ -77,12 +84,35 @@ func validateModelSpec(spec *ModelSpec) error {
 	if err := spec.Validate(); err != nil {
 		return err
 	}
+	if err := validateModelURI(spec.GetModelSourceUri()); err != nil {
+		return err
+	}
 	if err := validateReplicas(spec.MinReplicas, spec.MaxReplicas); err != nil {
 		return err
 	}
 	return nil
 }
 
+func validateModelURI(modelSourceURI string) error {
+	if modelSourceURI == "" {
+		return nil
+	}
+
+	// local path (not some protocol?)
+	if !regexp.MustCompile("\\w+?://").MatchString(modelSourceURI) {
+		return nil
```

Comment:
We gotta error if this fails, right?

Reply:
"Unknown model source"?

Reply:
so if MustCompile fails - it panics... if the regex match fails it means it is not a protocol.. so am assuming that means its some sort of local path :-)